RecoPixelVertexing/PixelTrackFitting/interface/BrokenLine.h

```diff
@@ -98,32 +98,44 @@ namespace brokenline {
                                                     double x0,
                                                     double y0,
                                                     Rfit::Matrix3d& jacobian) {
-    double tempA, tempU, tempB, tempC, deltaOrth, deltaPara, tempSmallU, xi, tempV, mu, lambda, zeta;
-    auto sinPhi = sin(circle.par(0));
-    auto cosPhi = cos(circle.par(0));
-    deltaPara = x0 * cosPhi + y0 * sinPhi;
-    deltaOrth = x0 * sinPhi - y0 * cosPhi + circle.par(1);
-    tempSmallU = 1 + circle.par(2) * circle.par(1);
-    tempC = -circle.par(2) * y0 + tempSmallU * cosPhi;
-    tempB = circle.par(2) * x0 + tempSmallU * sinPhi;
-    tempA = 2. * deltaOrth + circle.par(2) * (Rfit::sqr(deltaOrth) + Rfit::sqr(deltaPara));
-    tempU = sqrt(1. + circle.par(2) * tempA);
-    xi = 1. / (Rfit::sqr(tempB) + Rfit::sqr(tempC));
-    tempV = 1. + circle.par(2) * deltaOrth;
-    lambda = (0.5 * tempA) / (tempU * Rfit::sqr(1. + tempU));
-    mu = 1. / (tempU * (1. + tempU)) + circle.par(2) * lambda;
-    zeta = Rfit::sqr(deltaOrth) + Rfit::sqr(deltaPara);
-
-    jacobian << xi * tempSmallU * tempV, -xi * Rfit::sqr(circle.par(2)) * deltaOrth, xi * deltaPara, 2. * mu * tempSmallU * deltaPara, 2. * mu * tempV,
+
+    // Avoid multiple access to the circle.par vector.
+    using scalar = std::remove_reference<decltype(circle.par(0))>::type;
```

Comment:
why not "auto"?

Reply:
I wanted to make sure no float type crept in the computation (while opening the possibility to switch to float or quad), but I'm maybe not trusting the compiler enough.

Better just let it go "auto"?

Reply:
I would use `double` explicitly.